### PR TITLE
Clean up state-related terms

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Rename consensus state terms:
  - ConsensusStatus -> ConsensusStates
  - Consensus.status -> Consensus.state
  - Consensus.state -> Consensus.proposerStatus
  - Remove Consensus.setter 
- Rename transaction state terms:
  - TransactionStatue -> TransactionState
  - BLOCK -> IN_BLOCK, POOL -> IN_POOL, TIMEOUT -> TIMED_OUT, FAIL -> FAILED
- Rename client API:
  - /get_raw_consensus_state -> /get_raw_consensus_status
  - /get_consensus_state -> /get_consensus_status
- Rename JSON RPC API:
  - net_consensusState() -> net_consensusStatus()
  - net_rawConsensusState() -> net_rawConsensusStatus()

Related issue: https://github.com/ainblockchain/ain-blockchain/issues/452

### CODE REVIEW

File: consensus/index.js [hunk 1]
```diff
@@ -48,18 +48,17 @@ class Consensus {
...
+    this.setState(ConsensusStates.STARTING);
     this.consensusProtocolVersion = CONSENSUS_PROTOCOL_VERSION;
     this.majorConsensusProtocolVersion = VersionUtil.toMajorVersion(CONSENSUS_PROTOCOL_VERSION);
     this.epochInterval = null;
     this.startingTime = 0;
     this.timeAdjustment = 0;
     this.isReporting = false;
     this.isInEpochTransition = false;
-    this.state = {
+    this.proposerStatus = {
```
Comment (liayoo) [inline consensus/index.js:61]: Question) Maybe I misunderstood the new convention, but I feel like proposer**State** is more appropriate here? Could you explain why status is used here? Thanks!

Comment (liayoo) [reply consensus/index.js:61]: On second thought, I don't really see the benefits of using proposerStatus.. how about we just remove `proposerStatus` and use `this.epoch` and `this.proposer`? If you leave a TODO for me, I can take care of that :)

File: consensus/index.js [hunk 2]
```diff
@@ -1136,7 +1147,7 @@ class Consensus {
    */
   getRawState() {
```
Comment (liayoo) [inline consensus/index.js:1148]: Should this be getRawStatus()?

File: consensus/index.js [hunk 3]
```diff
@@ -1168,13 +1179,14 @@ class Consensus {
       health = false;
```
Comment (liayoo) [inline consensus/index.js:1179]: Should this be getStatus() ?

File: consensus/index.js (hunk 1 above)
Comment (platfowner) [reply consensus/index.js]: Done. It looks much cleaner. Thanks!

File: consensus/index.js (hunk 2 above)
Comment (platfowner) [reply consensus/index.js]: Fixed.

File: consensus/index.js (hunk 3 above)
Comment (platfowner) [reply consensus/index.js:1172]: Fixed.

**Comment by platfowner** [comment]
All comments addressed. PTAL! @liayoo

File: consensus/index.js [hunk 4]
```diff
@@ -216,9 +213,9 @@ class Consensus {
           `Discard the consensus message.`);
       return;
     }
-    if (this.status !== ConsensusStatus.RUNNING) {
-      logger.debug(`[${LOG_HEADER}] Consensus status (${this.status}) is not RUNNING ` +
-          `(${ConsensusStatus.RUNNING})`);
+    if (this.state !== ConsensusStates.RUNNING) {
+      logger.debug(`[${LOG_HEADER}] Consensus status (${this.state}) is not RUNNING ` +
```
Comment (liayoo) [inline consensus/index.js:217]: status -> state?

File: client/index.js [hunk 5]
```diff
@@ -435,16 +435,16 @@ app.get('/get_sharding', (req, res, next) => {
     .end();
 });
 
-app.get('/get_raw_consensus_state', (req, res) => {
-  const result = p2pServer.consensus.getRawState();
+app.get('/get_raw_consensus_status', (req, res) => {
```
Comment (liayoo) [inline client/index.js:438]: 👍 👍

**Review by liayoo — APPROVED** [review]
LGTM with a minor comment. Thanks for the cleanup!

File: consensus/index.js (hunk 4 above)
Comment (platfowner) [reply consensus/index.js]: Fixed. Thanks!

**Comment by platfowner** [comment]
@cshcomcom Please be noticed with the API change for the next data push.

**Comment by platfowner** [comment]
Thanks for the review! Now merging..

**Pull request merged**